catkit/emulators/iris_ao_controller.py

```diff
@@ -0,0 +1,111 @@
+import catkit.hardware.iris_ao.iris_ao_controller
+import catkit.hardware.iris_ao.util
+from catkit.interfaces.Instrument import SimInstrument
+
+
+class PoppyIrisAODM(poppy.dms.HexSegmentedDeformableMirror):
+
+    @property
+    def number_of_segments(self):
+        return self.dm_shape
+
+    def __init__(self, mcf_filename, custom_flat_filename, **super_kwargs):
+        super().__init__(**super_kwargs)
+
+        self.mirror_serial = None
+
+        # Read the manufacturers .mcf "flat" file.
+        # This is implicitly applied by the driver/controller and is NOT a present contribution of the command sent to
+        # the driver/controller by catkit.hardware.iris_ao.iris_ao_controller.IrisAoDmController.
+        self.mcf_filename = mcf_filename
+        #mcf_data, self.mirror_serial = catkit.hardware.iris_ao.util.read_mcf(self.mcf_filename, self.number_of_segments)
+        #mcf_relaxed_poppy_surface = -self.convert_command_to_poppy_surface(mcf_data, _include_relaxation=False)
+
+        self.custom_flat_filename = custom_flat_filename
+        custom_flat_data = catkit.hardware.iris_ao.util.read_ptt111(self.custom_flat_filename, self.number_of_segments)
+        custom_flat_relaxed_poppy_surface = -self.convert_command_to_poppy_surface(custom_flat_data, _include_relaxation=False)
+
+        # A flat Poppy surfaces := 0.
+        self.relaxed_poppy_surface = custom_flat_relaxed_poppy_surface  # + mcf_relaxed_poppy_surface
+        self.relax()  # ??? See https://github.com/spacetelescope/catkit/issues/63 (we don't currently relax the bostons like this).
+
+    def convert_command_to_poppy_surface(self, _include_relaxation=True):
+        # _include_relaxation exists only such that this func can be used to get the relaxed surface in the first place.
+
+        # pseudocode...
+        poppy_surface = # convert command to surface.
```

Comment:
@ivalaginja This also needs implementing.

Reply:
yup we already have this, I'll add it here